### PR TITLE
 plan: join's default value length should be changed when do column pruning.

### PR DESCRIPTION
**Pull request by winoros**
After column pruning, the `LogicalJoin.DefaultValues` may not be expected, and cause bug when do merge join

### CODE REVIEW

File: plan/column_pruning.go [hunk 1]
```diff
@@ -261,6 +262,11 @@ func (p *LogicalJoin) PruneColumns(parentUsedCols []*expression.Column) {
 	rChild := p.children[1].(LogicalPlan)
 	lChild.PruneColumns(leftCols)
 	rChild.PruneColumns(rightCols)
+	if p.JoinType == LeftOuterJoin {
+		p.DefaultValues = make([]types.Datum, p.children[1].Schema().Len())
```
Comment (hanfei1991) [inline plan/column_pruning.go]: DefaultValues = DefaultValues[:p.children[1].Schema().Len()] ?

Comment (winoros) [reply plan/column_pruning.go]: It seems that this will cause panic, i'll look into this tomorrow.

**Comment by hanfei1991** [comment]
LGTM

File: plan/column_pruning.go (hunk 1 above)
Comment (winoros) [reply plan/column_pruning.go]: This is caused by `USING` clause in join which may make the schema bigger than before.

**Pull request merged**